pkg/asset/installconfig/basedomain.go

```diff
@@ -48,9 +50,18 @@ func (a *baseDomain) Generate(parents asset.Parents) error {
 		}
 		a.BaseDomain = zone.Name
 		return platform.Azure.SetBaseDomain(zone.ID)
+	case gcp.Name:
+		var err error
+		a.BaseDomain, err = gcpconfig.GetBaseDomain(platform.GCP.ProjectID)
+		if err != nil && !(gcpconfig.IsForbidden(err) || gcpconfig.IsThrottled(err)) {
```

Comment:
this can probably look like https://github.com/openshift/installer/blob/700b75a3e2b88b7236498e0ba00708016bbc4f52/pkg/asset/installconfig/basedomain.go#L40-L43